kafka.go

```diff
@@ -31,17 +31,40 @@ type KafkaClient struct {
 	syncProducer  sarama.SyncProducer
 }
 
+// KafkaConfig is Kafka configuration to wait for a successful metadata response
+type KafkaConfig struct {
+	DialTimeout          time.Duration
+	ReadTimeout          time.Duration
+	WriteTimeout         time.Duration
+	MetadataRetryMax     int
+	MetadataRetryBackoff time.Duration
+}
+
 // NewKafkaClient creates new client to publish event to kafka
-func NewKafkaClient(realm string, brokerList []string) (*KafkaClient, error) {
-	configAsync := sarama.NewConfig()
+// The first producerConfiguration is the asynchronous producer configuration
+// meanwhile the second is the synchronous one
+func NewKafkaClient(realm string, brokerList []string, producerConfiguration ...*KafkaConfig) (*KafkaClient, error) {
+
+	var configAsync, configSync *sarama.Config
+	switch {
+	case len(producerConfiguration) > 1:
+		configAsync = configureKafka(producerConfiguration[0])
+		configSync = configureKafka(producerConfiguration[1])
+	case len(producerConfiguration) == 1:
+		configAsync = configureKafka(producerConfiguration[0])
+		configSync = sarama.NewConfig()
```

Comment:
Instead of just create a new config, I think we need a default value for KafkaConfig

Reply:
it is actually creating config with default value

Reply:
I mean default value that we defined for event stream so we dont use default value from sarama

Reply:
the purpose to make these configurable is to replace default value from sarama so I don't think we need default value set in this library too.

Reply:
oh i see, its okay for me